pkg/manifest/objects_test.go

```diff
@@ -0,0 +1,249 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package manifest
+
+import "testing"
+
+func TestHash(t *testing.T) {
+	hashTests := []struct {
+		kind      string
+		namespace string
+		name      string
+		want      string
+	}{
+		{"aaa", "nnn", "mmm", "aaa:nnn:mmm"},
```

Comment:
Optional: I get the rationale for testing with arbitrary strings, but in this case consider using strings like Deployment:istio-system:istio-pilot so it's self-documenting to users of the function.

Reply:
Done.